Failed C++ build.

Compiled using PlatformIO C++ for ESP32: toolchain: xtensa32 2.50200.80

I got a conflicting declaration linker error.

I got a good build by removing the declaration of dump_hex() from address.h.

Build Error:
.pio/libdeps/ttgo-t-watch/iota_client/src/core/address.h:64:47: error: conflicting declaration of 'void dump_hex(const byte_t*, size_t)' with 'C' linkage
void dump_hex(byte_t const data[], size_t len);
^
In file included from .pio/libdeps/ttgo-t-watch/iota_client/src/client/api/v1/get_health.h:11:0,
from src\SD_Object_Iota.h:15,
from src\SD_Main.h:24,
from src\SD_Object_FrameTimer.cpp:9:
.pio/libdeps/ttgo-t-watch/iota_client/src/core/types.h:21:20: note: previous declaration with 'C++' linkage
static inline void dump_hex(byte_t const data[], size_t len)

--- a/src/client/network/http_esp32.c
+++ b/src/client/network/http_esp32.c
@@ -70,6 +70,7 @@ static esp_err_t http_event_handler(esp_http_client_event_t* evt) {
   return ESP_OK;
 }
 
+#if 0
 static void http_download_chunk(void) {
   esp_http_client_config_t config = {
       .url = "http://httpbin.org/stream-bytes/8912",
@@ -83,6 +84,7 @@ static void http_download_chunk(void) {
   }
   esp_http_client_cleanup(client);
 }
+#endif
 
 static void init_config(esp_http_client_config_t* esp, http_client_config_t const* const conf) {
   esp->url = conf->url;
@@ -102,7 +104,7 @@ void http_client_init() {}
 void http_client_clean() {}
 
 int http_client_post(http_client_config_t const* const config, byte_buf_t const* const request,
-                     byte_buf_t* const response) {
+                     byte_buf_t* const response, long* status) {
   int ret = 0;
   esp_http_client_config_t esp_client_conf = {0};
   init_config(&esp_client_conf, config);
@@ -118,11 +120,12 @@ int http_client_post(http_client_config_t const* const config, byte_buf_t const*
     ESP_LOGE(TAG, "HTTP POST request failed: %s", esp_err_to_name(err));
     ret = -1;
   }
+  *status = esp_http_client_get_status_code(client);
   esp_http_client_cleanup(client);
   return ret;
 }
 
-int http_client_get(http_client_config_t const* const config, byte_buf_t* const response) {
+int http_client_get(http_client_config_t const* const config, byte_buf_t* const response, long* status) {
   int ret = 0;
   esp_http_client_config_t esp_client_conf = {0};
   init_config(&esp_client_conf, config);
@@ -136,6 +139,7 @@ int http_client_get(http_client_config_t const* const config, byte_buf_t* const 
     ret = -1;
   }
 
+  *status = esp_http_client_get_status_code(client);
   esp_http_client_cleanup(client);
   return ret;
 }

--- a/src/core/models/message.c
+++ b/src/core/models/message.c
@@ -125,7 +125,6 @@ int core_message_sign_transaction(core_message_t* msg) {
     }
   }
 
-end:
   if (unlocked_sig) {
     unlock_sig_free(&unlocked_sig);
   }

--- a/src/core/models/payloads/transaction.c
+++ b/src/core/models/payloads/transaction.c
@@ -15,8 +15,7 @@
 #define SIGNATURE_SERIALIZE_BYTES (1 + (1 + ED_PUBLIC_KEY_BYTES + ED_SIGNATURE_BYTES))
 
 static int byte_cmp(const byte_t* p1, const byte_t* p2, size_t len) {
-  byte_t b1, b2;
-  size_t count = 0;
+  byte_t b1 = 0, b2 = 0;
   for (size_t i = 0; i <= len; i++) {
     b1 = p1[i];
     b2 = p2[i];

--- a/src/wallet/wallet.c
+++ b/src/wallet/wallet.c
@@ -71,10 +71,14 @@ static int validate_pib44_path(char const path[]) {
 
 // get path from address
 static char* wallet_path_from_index(iota_wallet_t* w, uint32_t index) {
+  int ret_size = 0;
   char* path = malloc(IOTA_ACCOUNT_PATH_MAX);
   if (path) {
     // Bip44 Paths: m/44'/4128'/Account'/Change'/Index'
-    snprintf(path, IOTA_ACCOUNT_PATH_MAX, "%s/%" PRIu32 "'", w->account, index);
+    ret_size = snprintf(path, IOTA_ACCOUNT_PATH_MAX, "%s/%" PRIu32 "'", w->account, index);
+    if (ret_size >= IOTA_ACCOUNT_PATH_MAX) {
+      path[IOTA_ACCOUNT_PATH_MAX - 1] = '\0';
+    }
   }
   return path;
 }

